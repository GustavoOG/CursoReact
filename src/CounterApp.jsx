import React, { useState } from "react";
import Proptypes from "prop-types";

export const CounterApp = ({ value }) => {

  console.log('render');
  const [counter, setCounter] = useState(value);

  const handleAdd = () => {
    //console.log(event);
    setCounter(counter + 1);
    //setCounter((c) => c + 1);
  };

  const handlerSubstract = () => setCounter(counter - 1);
  const handlerReset = () => setCounter(value);

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>
      <button onClick={handleAdd}>+1</button>
      <button onClick={handlerSubstract}>-1</button>
      <button aria-label="btn-reset" onClick={handlerReset}>Reset</button>
    </>
  );
};

CounterApp.prototype = {
  value: Proptypes.number.isRequired,
};
