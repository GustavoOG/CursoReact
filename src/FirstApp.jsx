import React from "react";
import { Fragment } from "react";
import PropTypes from "prop-types";
//prop-types para definir tipo

const newMessage = {
  message: "Hola mundo",
  title: "Gustavo",
};

const getResult = (a, b) => {
  return a + b;
};

export const FirstApp = ({ title, subTitle, name }) => {
  //console.log(props);
  //   if (!title) {
  //     throw new Error("El title no existe");
  //   }

  return (
    //<Fragment> <-- fragmento
    <>
      <h1 data-testid="test-title"> {title} </h1>
      {/* <h1>{getResult(3,4)}</h1> 
       <code>{JSON.stringify(newMessage)}</code>  */}
      <p>{subTitle}</p>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
  title: "Hola, Soy Goku",
  subTitle: "Soy un subtitulo",
  name: "Gustavo Ortiz!!",
};
