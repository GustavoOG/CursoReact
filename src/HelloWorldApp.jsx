// function App() {
//     return <h1>Hola Mundo!!!!</h1>;
//   }

//   export default App;



//rafc
// import React from 'react'

export const HelloWorldApp = () => {
  return (
    <h1>Hola mundo!!!</h1>
  )
}


//export const HelloWorldApp =() =>  <h1>Hola mundo!!!</h1>;