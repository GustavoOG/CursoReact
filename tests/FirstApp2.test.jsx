import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe("Pruebas en <FirstApp/>", () => {
  const title = "Hola, soy Goku";
  const subTitle = "Soy un subtitulo";
  test("debe de hacer match con el snapshopt", () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  });

  test('debe de ostrar el mensjae "Hola, soy Goku"', () => {
    render(<FirstApp title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
    //screen.debug();
  });

  test("debe de mostrar el titulo en h1", () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      title
    );
  });

  test('debe de notrar el subtitulo enviado por props', () => { 
    render(<FirstApp title={title}  subTitle={subTitle}/>);
    expect(screen.getAllByText(subTitle).length).toBe(2);
   })
});
