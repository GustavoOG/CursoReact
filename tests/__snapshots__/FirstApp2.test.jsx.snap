// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp/> debe de hacer match con el snapshopt 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, soy Goku
     
  </h1>
  <p>
    Soy un subtitulo
  </p>
  <p>
    Soy un subtitulo
  </p>
  <p>
    Gustavo Ortiz!!
  </p>
</div>
`;
